django/db/backends/sqlite3/base.py

```diff
@@ -247,11 +247,10 @@ def _set_autocommit(self, autocommit):
             self.connection.isolation_level = level
 
     def disable_constraint_checking(self):
-        if self.in_atomic_block:
-            # sqlite3 cannot disable constraint checking inside a transaction.
-            return False
-        self.cursor().execute('PRAGMA foreign_keys = OFF')
-        return True
+        with self.cursor() as cursor:
+            cursor.execute('PRAGMA foreign_keys = OFF')
+            enabled = cursor.execute('PRAGMA foreign_keys').fetchone()[0]
```

Comment:
Maybe a comment here? 

`...foreign key constraints cannot be disabled within open transaction...` or similar — this is Dark Arts here, any signposts for the future handy. 🙂

Reply:
Makes sense.